src/routes/strange-router.js

```diff
@@ -0,0 +1,99 @@
+const React = require('react');
+const T = require('prop-types');
+
+const { default: MatchPath } = require('react-router-dom/matchPath');
+const { default: Router } = require('react-router-dom/Router');
+const { default: Switch } = require('react-router-dom/Switch');
+const { default: Route } = require('react-router-dom/Route');
+
+// Ensure we're using the exact code for default root match
+
+const { computeMatch } = Router.prototype;
+
+exports.Router = class StrangeRouter extends React.PureComponent {
+
+    static propTypes = {
+        history: T.object,
+        routes: T.array.isRequired
+    }
+
+    constructor() {
+
+        super();
+        this._renderRoutes = this.renderRoutes.bind(this);
+    }
+
+    renderRoutes(routes) {
+
+        const renderRouteRecursive = (route, i) => {
+
+            let path = route.path;
+            if (path[0] !== '/') {
+                path = `/${path}`;
+            }
+
+            return (
+                <Route
+                    exact={route.exact}
+                    key={i}
```

Comment:
I agree, the react training site used index as the key I was like =/ but still used it loL. React must have a key generator internally, otherwise should we write a keyGen func?

Reply:
Perhaps it can be based on the path